### PR TITLE
[FrameworkBundle][SecurityBundle] Moved security expression providers pass logic to SecurityBundle

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

**Comment by stof** [comment]
We probably need to make SecurityBundle conflict with older versions of FrameworkBundle to ensure we don't end up with double registration

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AddExpressionLanguageProvidersPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('security.expression_language')) {
```
Comment (iltar) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php]: This can be a guard clause now

**Review by linaori — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php (hunk 1 above)
Comment (HeahDude) [reply src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php]: If the code remains cleans and small with one indent, I would keep a readable true condition here. I'll wait for other opinions before changing the style.

**Comment by HeahDude** [comment]
@stof I added a BC layer in https://github.com/symfony/symfony/pull/27611/commits/50a09da0310ba4865ffab93d7ccd7eb85a7ec950. Thanks for the review.

**Comment by HeahDude** [comment]
Failures unrelated.

Status: needs review

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/composer.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "conflict": {
         "symfony/var-dumper": "<3.4",
         "symfony/event-dispatcher": "<3.4",
-        "symfony/framework-bundle": "<4.1.1",
+        "symfony/framework-bundle": "<4.2",
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: L33 should bump to 4.2 also

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php [hunk 3]
```diff
@@ -22,6 +23,17 @@
  */
 class AddExpressionLanguageProvidersPass implements CompilerPassInterface
 {
+    private $handleSecurityLanguageProviders;
+
+    public function __construct(bool $handleSecurityLanguageProviders = true)
+    {
+        if ($handleSecurityLanguageProviders) {
+            @trigger_error(sprintf('Registering services tagged with "security.expression_language_provider" with "%s" is deprecated since Symfony 4.2, use the "%s" instead.', __CLASS__, SecurityExpressionLanguageProvidersPass::class), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php]: I'd suggest removing the first "with".

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ CHANGELOG
 
  * Allowed configuring taggable cache pools via a new `framework.cache.pools.tags` option (bool|service-id)
  * Deprecated auto-injection of the container in AbstractController instances, register them as service subscribers instead
+ * Deprecated the processing of services tagged `security.expression_language_provider` in favor of the new SecurityBundle pass
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: Deprecated processing of services tagged `security.expression_language_provider` in favor of a new `AddExpressionLanguageProvidersPass` in SecurityBundle.

File: src/Symfony/Bundle/SecurityBundle/composer.json (hunk 2 above)
Comment (HeahDude) [reply src/Symfony/Bundle/SecurityBundle/composer.json]: Fixed.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php (hunk 3 above)
Comment (HeahDude) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php]: Agreed, done.

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md (hunk 4 above)
Comment (HeahDude) [reply src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: Done.

**Comment by nicolas-grekas** [comment]
Thank you @HeahDude.

**Pull request merged**